### ISSUE TITLE
Support vim-clap as an alternative to coc-list

### ISSUE THREAD

**Issue by vn-ki** [neoclide/coc.nvim#1732]
**Is your feature request related to a problem? Please describe.**
IMHO, coc-list is not as polished or feature rich as some other fuzzy matchers. I understand that replacing coc-list with fzf completely would be near impossible, but I recently switched to [vim-clap](https://github.com/liuchengxu/vim-clap) and it is friggin amazing, both in speed and usability.

One of the arguments against fzf was that it wouldn't support immediate feedback on typing, but clap can handle this ([I think](https://github.com/liuchengxu/vim-clap/blob/master/PROVIDER.md), see [grep](https://github.com/liuchengxu/vim-clap/blob/master/autoload/clap/provider/grep.vim)). It also looks much more modern that coc-list or FZF.

**Describe the solution you'd like**
I would like to see support for vim-clap as an alternative to coc-list. 

Pinging @liuchengxu

Related #926

**Comment by liuchengxu** [comment]
Thanks, @vn-ki . I believe vim-clap already has all the basic APIs to implement coc-clap. Just let me know if someone finds vim-clap misses some feature when working on this.

I won't do it myself as I actually have a new unpublished plugin which is inspired by https://github.com/emacs-lsp/lsp-ui. I'll use `references/definition` provided by LSP/grep in that plugin, it will also be used to replace some listing/viewing functions in coc.nvim.

<img width="1680" alt="截屏2020-04-08 下午10 01 55" src="https://user-images.githubusercontent.com/8850248/78793325-d3bb6700-79e4-11ea-9256-38db82dc44e0.png">

**Comment by vn-ki** [comment]
@liuchengxu Wow. That looks great. Please release it soon, I'm excited to try it out (I'm fan of lsp-ui too).

**Comment by vn-ki** [comment]
@liuchengxu Is it possible to add custom actions to clap? Like below?

![image](https://user-images.githubusercontent.com/31964688/79130239-41d8a300-7dc4-11ea-8243-1f2b12e5a0e9.png)

**Comment by liuchengxu** [comment]
@vn-ki That's on the TODO list, see https://github.com/liuchengxu/vim-clap/issues/51#issuecomment-596211333.

**Comment by vn-ki** [comment]
I decided to modify coc-fzf to use clap. You can see the result here. Most of it works, waiting on actions for coc_extensions.

https://github.com/vn-ki/coc-clap

**Comment by liuchengxu** [comment]
@vn-ki You can try https://github.com/liuchengxu/vim-clap/pull/396 for implementing the actions for cocc_extensions. I've added an example there.

**Comment by ptn** [comment]
@vn-ki 

> I understand that replacing coc-list with fzf completely would be near impossible

Curious, why is that?

**Comment by vn-ki** [comment]
Because fzf does not provide a way for callback on type. FZF is just a terminal program running in terminal buffer.

**Comment by ptn** [comment]
Ah, callback on type is not available, I don't think. There's only --bind and --expect.

Thanks.

**Comment by chemzqm** [comment]
Use https://github.com/vn-ki/coc-clap
